packages/langium/src/parser/lexer.ts

```diff
@@ -66,6 +67,45 @@ export class DefaultLexer implements Lexer {
     }
 }
 
+/**
+ * A lexer that is aware of indentation in the input text.
+ * The only purpose of this lexer is to reset the internal state of the {@link IndentationAwareTokenBuilder}
+ * between the tokenization of different text inputs.
+ *
+ * In your module, you can override the default lexer with this one as such:
+ * ```ts
+ * parser: {
+ *    TokenBuilder: () => new IndentationAwareTokenBuilder(),
+ *    Lexer: (services) => new IndentationAwareLexer(services),
+ * }
+ * ```
+ */
+export class IndentationAwareLexer extends DefaultLexer {
+    private indentationTokenBuilder?: IndentationAwareTokenBuilder;
```

Comment:
Suggestion: With the way our service implementations are designed (i.e. being capable of overriding everything), we don't keep `private` methods/fields. Please mark them all as `protected`. This also applies to the other file. Also, make this field `readonly`.